### ISSUE TITLE
[Feature Request]: Setting to Add Videos to Top of Custom-sorted Playlist

### ISSUE THREAD

**Issue by deepspaceaxolotl** [FreeTubeApp/FreeTube#8007]
### Guidelines

- [x] I have [searched the issue tracker for open and closed issues](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+sort%3Arelevance-desc) that are similar to the feature request I want to file, without success.
- [x] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the feature request I want to file, without success.
- [x] This issue contains only one feature request.
- [x] I have read and agree to follow the [rules](https://docs.freetubeapp.io/community/rules/).

### Problem Description

Currently, adding a video to a playlist that has custom sorting set adds it to the bottom of the playlist. This can make it very annoying to get a video to the top if there are dozens or even hundreds of videos, requiring many clicks (which can be an accessibility issue for people with disabilities like mine).

My use case is that I used to have a reverse-chronologically sorted playlist for my watch later list, with new stuff getting added to the top and older stuff that I hadn't watched yet because I wasn't ready to getting filtered down to the bottom, creating a natural hierarchy, with stuff that I wanted to watch sooner at the top and easily accessible. I've recently switch to custom sorting for more organisation and am now running into this issue interrupting my desired workflow.

### Proposed Solution

A setting (probably best per-playlist) to add new videos to the top or bottom of the playlist when custom sorting is in use.

### Alternatives Considered

Better sorting features that would let one easily move a video to the top of the playlist (will likely open an issue about this regardless if there isn't one already).

### Issue Labels

ease of use improvement

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Weird we had an issue open for this https://github.com/FreeTubeApp/FreeTube/issues/4603 but for some reason i closed it. It also had an accompanying PR that didnt get merged because it had some issues https://github.com/FreeTubeApp/FreeTube/pull/4604

**Comment by deepspaceaxolotl** [comment]
I hadn't seen that! That could definitely work as a short-term fix (and be very useful for resorting videos in general!), though for longer-term custom sort use in one's main watch later playlist, I think my proposal would still be a better experience, so that videos don't have to constantly be sent to the top after getting added.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Im sorry but your proposed solution is too niche. I will reopen #4603 as that is the same as the alternative you suggested and is more realistic to implement

**Comment by deepspaceaxolotl** [comment]
Fair. I think it does add more work when adding many videos to a custom-sorted playlist (such as one being used as a watch later playlist), but certainly better than the status quo.

**Comment by deepspaceaxolotl** [comment]
Running into issues around this still, particularly when adding multiple videos to a playlist at once. While that alternative issue would be much better than the status quo, it would still add an extra click for every video added (and if the page scrolls up to where the video has gone after clicking it, a lot more scrolling to do it for multiple videos; an extra click for every video adds up pretty quickly, as do scrolls, and that can suck for those of us with chronic pain).

Outside of letting one select videos to move to the top/bottom as a block, another alternative I can think of to my original suggestion is a checkbox when selecting the playlist for a video to go to that will put the video at the top of whatever playlist is selected, remembered until the checkbox is unselected again and applying to the bookmarked playlist, as well. This would avoid needing to save that setting on a per-playlist basis.

The action flow would be something like this: use the + button to select a playlist to add the video to > check the checkbox for it to go to the top > select playlist or close the popup > videos added to custom-sorted playlists are added to the top, whether via the + button or the bookmarked playlist button > open the popup again with the + button to switch the checkbox off when one no longer wants to add videos to the top of custom-sorted playlists.

(Alternatively to this, a settings-page setting for where videos should go for custom-sorted playlists (bottom or top) would also work, just with a slightly less intuitive flow, but a fine compromise to have this kind of behaviour at all.)

**Comment by deepspaceaxolotl** [comment]
Given that the lack of this feature still impacts my use daily (having to move every new video added to a playlist to the top just because it's a custom-sorted playlist), I'd like to know if any of my proposed solutions could be implemented? At the end of the day, I'd just like to be able to put things at the top of my Watch Later playlist (like the default behaviour for it on YouTube) while still being able to move videos around manually, without too much fuss. I can imagine many people who would appreciate being able to keep that behaviour of a Watch Later–type playlist while enabling custom sorting.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> like the default behaviour for it on YouTube

From my testing this is certainly not the case on a custom sorted playlist. New videos are always added to the bottom of the list. Please provide a video that shows the behavior from your side.

**Comment by deepspaceaxolotl** [comment]
This is specific to the Watch Later playlist, which is the behaviour/usecase I'm hoping to replicate with improvements (custom sorting), not other playlists as far as I'm aware. I'll see if I can record a video somehow soon and share it here.

(I also want to apologise for my replies on several issues earlier; I was going through my subscribed issues and thought to check in on some, including some FreeTube ones, but the last thing I wanted was to be a nuisance, especially to people working as hard as you are. I'm very sorry, and I'll do what I can to avoid troubling you all in the future.)
